### ISSUE TITLE
[HybridApp] Problem #3: Concierge bubble UI should direct the user to the New experience 

### ISSUE THREAD

**Issue by AndrewGable** [Expensify/App#37856]
## Problem
On OldWeb, when you click the Concierge button you are directed to the New experience opened to your chat with Concierge. This is a great first experience to get to use our new chat functionalities with our support team. We would like to match that user experience with the Old experience, when the user clicks the Concierge button we will replace the entire page and navigation stack with the New experience with the Concierge chat opened.

Note: We will only redirect users who have been chosen for a migration to Concierge on the New experience (i.e. users with a classicRedirect value set). This is because we do not want to confuse Collect users talking to Concierge if they have not chosen to use the New experience as they might not know how to “Return to Expensify Classic”.

## Solution

Within the [ConciergeManager.js](https://github.com/Expensify/Mobile-Expensify/blob/503b8a2120edf1dc44740d961d0fdf33d847efdb/app/libs/ConciergeManager.js) presentConciergeChat function, launch a NewDot deep link if the ‘classicRedirect’ NVP exists for the user: 

    ...
    presentConciergeChat() {
        const user = Expensify.getInstance().getUser();
        let tryNewDotNVP = null;
        if (user.getNameValuePair('tryNewDot')) {
            tryNewDotNVP = user.getNameValuePair('tryNewDot');
        }

        if (tryNewDotNVP && tryNewDotNVP['classicRedirect'] && !tryNewDotNVP['classicRedirect']['dismissed']) {
            // launch NewDot
      // TODO in next code block change
            return;
        }

        // Else, continue launching the native Concierge Dialog
        ...
    ...

We launch NewDot with a deep link URL after retrieving an auth token (changes in ):

    ...
    presentConciergeChat() {
        const user = Expensify.getInstance().getUser();
        let tryNewDotNVP = null;
        if (user.getNameValuePair('tryNewDot')) {
            tryNewDotNVP = user.getNameValuePair('tryNewDot');
        }

        if (tryNewDotNVP && tryNewDotNVP['classicRedirect']) {
            redirectToNewDotConcierge();
            return;
        }
        ...

    redirectToNewDotConcierge() {
        API.GetShortLivedAuthToken().done(function (response) {
            const conciergeAccountID = PersonalDetailsUtils.getAccountIDsByLogins([CONST.EMAIL.CONCIERGE]);
            const chatId = ReportUtils.getChatByParticipants(conciergeAccountID);
            const URL = Expensify.getInstance().buildNewDotTransitionURL(response, chatId);
            yaplOpenNewApp(URL);
        });
    }

**Comment by Julesssss** [comment]
Holding this one for now, as I'm unable to test the redirect due to authentication. [Here](https://github.com/Expensify/Expensify/issues/371824#issuecomment-1976828017) are the latest documented steps.

**Comment by Julesssss** [comment]
I'm unblocked from the auth issue now. But due to prioritisation I need to work on a critical notification issue before picking up this task

**Comment by Julesssss** [comment]
I'll get to this early May

**Comment by Julesssss** [comment]
Still aiming for early May, once I have finished the post- redirect fire  tasks

**Comment by Julesssss** [comment]
Dealing with high priority post-fire chores to improve slow API queries. This doesn't seem high priority right now as I haven't seen much movement with the hybrid app lately.

**Comment by Julesssss** [comment]
I'll get this done soon. I need to finish other more important slow API tasks

**Comment by Julesssss** [comment]
Same as above, getting closer

**Comment by staszekscp** [comment]
Hey! I may take care of that, since I have some free time right now :)

**Comment by staszekscp** [comment]
Hey! I wanted to tackle this issue today, but I noticed a small problem: 

After the changes from [this PR](https://github.com/Expensify/Mobile-Expensify/pull/12688) if we are on OldDot the `classicRedirect` value will always be `false`, therefore we'll always open the current Concierge modal. On the other hand, if the `classicRedirect` is `true`, we will always open NewDot, so we won't be able to press the concierge bubble 😄 The only way to execute this code would be to open NewDot, then press the `Switch back to Expensify Classic`, and use it then _within one session._

That's how I also noticed that while closing NewDot we do not change the `classicRedirect` value, because on following launches we always display NewDot - is that intentional?
